packages/renderer/src/lib/kube/KubeEditYAML.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+import MonacoEditor from '../editor/MonacoEditor.svelte';
+import Button from '../ui/Button.svelte';
+
+// Make sure that when using the MonacoEditor, the content is "stringified" before
+// being passed into this component. ex. stringify(kubeDeploymentYAML)
+// this is the 'initial' content.
+export let content = '';
+let key = 0; // Initial key
+let originalContent = '';
+let editorContent: string;
+let inProgress = false;
+
+// Reactive statement to update originalContent only if it's blank and content is not
+// as sometimes content is blank until it's "loaded". This does not work with onMount,
+// so we use the reactive statement
+$: if (originalContent === '' && content !== '') {
+  originalContent = content;
+}
+
+// Handle the content change from the MonacoEditor / store it for us to use for applying to the cluster.
+function handleContentChange(event: CustomEvent<string>) {
+  editorContent = event.detail;
+}
+
+/////////////
+// Buttons //
+/////////////
+
+// Function that will update content with originalContent
+// when 'revertChanges' button is pressed.
+async function revertChanges() {
+  content = originalContent;
+  key++; // Increment the key to force re-render
+}
+
+async function applyToCluster() {
+  // Get the current context name
+  let contextName = await window.kubernetesGetCurrentContextName();
+  if (!contextName) {
+    return;
+  }
+
+  try {
+    inProgress = true;
+    await window.kubernetesApplyResourcesFromYAML(contextName, editorContent);
+    await window.showMessageBox({
+      title: 'Kubernetes',
+      type: 'info',
+      message: 'Succesfully applied Kubernetes YAML',
+      buttons: ['OK'],
+    });
+  } catch (error) {
+    console.error('error playing kube file', error);
+    await window.showMessageBox({
+      title: 'Kubernetes',
+      type: 'error',
+      message: 'Could not apply Kubernetes YAML: ' + error,
```

Comment:
I would use a dot(.) instead of colon(:), as error messages coming from Kubernetes seem to start with 'Error: '

```
Could not apply Kubernetes YAML: Error: resourceVersion should not be set on objects to be created
```
```suggestion
      message: 'Could not apply Kubernetes YAML. ' + error,
```